framework/src/base/NonlinearSystemBase.C

```diff
@@ -1850,11 +1852,11 @@ NonlinearSystemBase::computeJacobianInternal(SparseMatrix<Number> & jacobian, Mo
 
     // For the matrix in the right side of generalized eigenvalue problems, its conresponding
     // rows are zeroed if homogeneous Dirichlet boundary conditions are used.
-   if (kernel_type == Moose::KT_EIGEN)
-     _fe_problem.assembly(0).zeroCachedJacobianContributions(jacobian);
-   // Set the cached NodalBC values in the Jacobian matrix
-   else
-     _fe_problem.assembly(0).setCachedJacobianContributions(jacobian);
+    if (kernel_type == Moose::KT_EIGEN)
+      _fe_problem.assembly(0).zeroCachedJacobianContributions(jacobian);
+    // Set the cached NodalBC values in the Jacobian matrix
+    else
+      _fe_problem.assembly(0).setCachedJacobianContributions(jacobian);
```

Comment:
Is there a change you can move this into 98ac4f8e5a32fd7ac0336733107e6d6d7e2b8cbd where it belongs?